polyfill/spatial-navigation-polyfill.js

```diff
@@ -1174,31 +1177,46 @@
     const P1 = Math.abs(points.entryPoint[0] - points.exitPoint[0]);
     const P2 = Math.abs(points.entryPoint[1] - points.exitPoint[1]);
 
-    // A = The euclidian distance between P1 and P2.
+    // A: The euclidian distance between P1 and P2.
     const A = Math.sqrt(Math.pow(P1, 2) + Math.pow(P2, 2));
     let B, C;
 
-    // B: The absolute distance in the dir direction between P1 and P2, or 0 if dir is null.
-    // C: The absolute distance in the direction which is orthogonal to dir between P1 and P2, or 0 if dir is null.
+    // B: The absolute distance in the direction which is orthogonal to dir between P1 and P2, or 0 if dir is null.
+    // C: The intersection edges between a candidate and the starting point.
+
+    // D: The square root of the area of intersection between the border boxes of candidate and starting point
+    const intersectionRect = getIntersectionRect(rect1, rect2);
+    const D = intersectionRect.area;
+
     switch (dir) {
     case 'left':
       /* falls through */
     case 'right' :
-      B = P1;
-      // If not aligned => add bias
-      if (!isAligned(rect1, rect2, dir))
+      // If two elements are aligned, add align bias
+      if (isAligned(rect1, rect2, dir)) {
+        alignBias = intersectionRect.height / rect1.height;
+        if (alignBias > 1) align_bias = 1;
+      }
+      else  // else, add orthogonal bias
         orthogonalBias = (rect1.height / 2);
-      C = (P2 + orthogonalBias) * kOrthogonalWeightForLeftRight;
+
+      B = (P2 + orthogonalBias) * kOrthogonalWeightForLeftRight;
+
```

Comment:
The revised distance function is 

> "distance = euclidian + displacement - alignment - sqrt(Overlap)" 
> (https://drafts.csswg.org/css-nav-1/#calculating-the-distance)

Therefore, for aligning with the spec,
**B factor becomes the factor representing the displacement** (the distance on the axis orthogonal to navigation direction)
**C factor becomes the factor representing the alignment.**

It's bit confusing, yeah. 
Since the conventional C factor was representing the displacement, 
so this change is about switching the C factor  to B factor.